### PR TITLE
SE Config Profiles

### LINKED ISSUE #1416: Config Profiles
EDITED by @tomas-langer 
As agreed, the following config profile approach will be added to Helidon SE:

- [x] support inline config in meta config
- [x] support config.profile as a fallback system property to define the file (name config-profile-${profile}.xxx
- [x] support env var to define a different "profile" `HELIDON_CONFIG_PROFILE`
- [x] basically rename "meta configuration" to "config profile"
- [x] add a guide/example with a best practise
sources:
  - type: "file"
    properties:
      path: "config/application-dev.yaml"
  - type: "inlined"
    properties:
      key: "value"
      "server.port": 40877

Original text:
The idea behind config profiles is to be able to repeat configuration (properties) in such a way that they can be selected depending on the target environment in which the application will run. Selection of a profile can be done _externally_ via a system property or environment variable such as HELIDON_CONFIG_PROFILE.

~~As an example, different HTTP ports may be necessary in production and development:~~

```
%dev.http.port=8080
%prod.http.port=7001
```

~~where dev and prod are environment names. A property without a profile prefix would be active in all environments.~~

~~The notion of config profiles has been requested a few times, and it is supported by many other microservice frameworks.~~

### LINKED ISSUE #2609: Documentation Helidon SE: Polling strategy of type watch is not supported
Into Helidon SE Config Guide, section Reacting to configuration updates: When starting the application it throws

`Exception in thread "main" java.lang.IllegalArgumentException: Polling strategy of type watch is not supported. Supported types: [regular]
	at io.helidon.config.MetaProviders.lambda$pollingStrategy$16(MetaProviders.java:144)
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at io.helidon.config.MetaProviders.pollingStrategy(MetaProviders.java:144)
	at io.helidon.config.MetaConfig.pollingStrategy(MetaConfig.java:117)
	at io.helidon.config.ConfigValues.lambda$create$4(ConfigValues.java:146)
	at io.helidon.config.ConfigValues$GenericConfigValueImpl.asOptional(ConfigValues.java:228)
	at io.helidon.config.ConfigValue.ifPresent(ConfigValue.java:227)
	at io.helidon.config.AbstractSourceBuilder.config(AbstractSourceBuilder.java:89)
	at io.helidon.config.AbstractConfigSourceBuilder.config(AbstractConfigSourceBuilder.java:70)
	at io.helidon.config.FileConfigSource$Builder.config(FileConfigSource.java:205)
	at io.helidon.config.FileConfigSource.create(FileConfigSource.java:80)
	at io.helidon.config.MetaProviders$BuiltInConfigSourcesProvider.create(MetaProviders.java:289)
	at io.helidon.config.MetaProviders$BuiltInConfigSourcesProvider.create(MetaProviders.java:259)
	at io.helidon.config.MetaProviders.lambda$configSource$6(MetaProviders.java:116)
	at java.base/java.util.Optional.map(Optional.java:265)
	at io.helidon.config.MetaProviders.configSource(MetaProviders.java:116)
	at io.helidon.config.MetaConfig.configSource(MetaConfig.java:172)
	at io.helidon.config.BuilderImpl.lambda$config$3(BuilderImpl.java:361)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at io.helidon.config.BuilderImpl.lambda$config$4(BuilderImpl.java:361)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.helidon.config.ConfigValue.ifPresent(ConfigValue.java:227)
	at io.helidon.config.BuilderImpl.config(BuilderImpl.java:361)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.helidon.config.Config$Builder.metaConfig(Config.java:1566)
	at io.helidon.config.Config.create(Config.java:313)
	at io.helidon.examples.quickstart.se.Main.startServer(Main.java:41)
	at io.helidon.examples.quickstart.se.Main.main(Main.java:29)`

REPRODUCER:

```
mvn -U archetype:generate -DinteractiveMode=false \
    -DarchetypeGroupId=io.helidon.archetypes \
    -DarchetypeArtifactId=helidon-quickstart-se \
    -DarchetypeVersion=2.1.1-SNAPSHOT \
    -DgroupId=io.helidon.examples \
    -DartifactId=helidon-quickstart-se \
    -Dpackage=io.helidon.examples.quickstart.se
```

- Create `config-file.properties` with content `app.greeting=HelloFrom-config-file.properties` into /helidon-quickstart-se directory
- Create `meta-config.yaml` into /helidon-quickstart-se directory with this content:
```
sources:
  - type: "file"
    properties:
      path: "./config-file.properties"
      polling-strategy:
        type: "watch"
  - type: "classpath"
    properties:
      resource: "application.yaml"
```
- Update the GreetService class 
```
import java.util.function.Consumer;
...

    GreetService(Config config) {
        Config greetingConfig = config.get("app.greeting"); // (1)
        greeting.set(greetingConfig.asString().orElse("Ciao"));
        greetingConfig.onChange((Consumer<Config>) cfg -> greeting.set(cfg.asString().orElse("Ciao"))); // (2)
    }
```
- Build and run the application
```
mvn package -DskipTests=true
java -jar target/helidon-quickstart-se.jar
```

### PR DESCRIPTION
**Pull request by tomas-langer**
Resolves #1416 
Resolves #2609 

Support for configuration profiles (what we used to call meta configuration).
Example
Updated guide
Updated configuration docs
(Fixed all configuration docs links)

Profile can be switched by `config.profile` or `helidon.config.profile` system property, by `HELIDON_CONFIG_PROFILE` environment variable.

### CODE REVIEW

**Review by spericas — APPROVED** [review]

File: docs/se/config/09_config-profiles.adoc [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+Optional:
+
+* `delay` (`Duration`) - initial delay between retries
+* `delay-factor` (`double`) - `delay` is repeatedly multiplied by this each retry to compute
+the delay for each successive retry
+* `call-timeout` (`Duration`) - timeout for a single invocation to load the source
+* `overall-timeout` (`Duration`) - total timeout for all retry calls and delays
+|===
+
+To specify a custom polling strategy or custom retry policy, implement the interface
```
Comment (ljamen) [inline docs/se/config/09_config-profiles.adoc:295]: (Q) implement the interface? is that correct?

**Review by ljamen — COMMENTED** [review]
Small updates, one question re: 'implementing the interface'. Is that the correct terminology?

File: docs/se/config/09_config-profiles.adoc (hunk 1 above)
Comment (tomas-langer) [reply docs/se/config/09_config-profiles.adoc:295]: It is correct, but I clarified it to list the exact interfaces to implement.

**Review by ljamen — APPROVED** [review]
Changes look good, thanks.

**Pull request merged**